birdhouse/config/geoserver/config/proxy/conf.extra-service.d/geoserver.conf.template

```diff
@@ -1,5 +1,5 @@
     location /geoserver/ {
-        proxy_pass http://geoserver:8080/geoserver/;
+        proxy_pass http://${PAVICS_FQDN}${TWITCHER_PROTECTED_PATH}/geoserver/;
```

Comment:
Opps sorry, I mixed up the "verify/protected" path performance problem with JupyterHub and this one.

Maybe Geoserver won't be affected by the performance problem but just to prepare in advance, can this behavior be toggled off?  Meaning the redirect in the front proxy be default behind Twitcher but make it configurable.

Reply:
Should this `proxy_pass` directive use `https` since it is using `PAVICS_FQDN`?

Reply:
I've put the check behind the verify endpoint and introduced the `GEOSERVER_SKIP_AUTH` setting to skip checking twitcher entirely if needed. 

I've added comments in a few places to make it clear that we do not recommend using `GEOSERVER_SKIP_AUTH` since it opens up geoserver to full public access without any checks.